cdap-api/src/main/java/co/cask/cdap/api/flow/FlowSpecification.java

```diff
@@ -74,6 +76,21 @@
   List<FlowletConnection> getConnections();
 
   /**
+   * @return Map of streams and {@link StreamSpecification} created in this flow.
+   */
+  Map<String, StreamSpecification> getStreams();
```

Comment:
True. I will move these methods into the configurers and remove them from the Program specifications since there is no use case to expose this to the users. That will also take care of the repeated code in specifications.